libs/community/langchain_community/embeddings/clova.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import annotations
+
+from typing import Dict, List, Optional
+
+import requests
+from langchain_core.embeddings import Embeddings
+from langchain_core.pydantic_v1 import BaseModel, Extra, SecretStr, root_validator
+from langchain_core.utils import convert_to_secret_str, get_from_dict_or_env
+
+
+class ClovaEmbeddings(BaseModel, Embeddings):
+    """
+    Clova's embedding service.
+
+    To use this service,
+
+    you should have the following environment variables
+    set with your API tokens and application ID,
+    or pass them as named parameters to the constructor:
+
+    - ``CLOVA_EMB_API_KEY``: API key for accessing Clova's embedding service.
+    - ``CLOVA_EMB_APIGW_API_KEY``: API gateway key for enhanced security.
+    - ``CLOVA_EMB_APP_ID``: Application ID for identifying your application.
+
+    Example:
+        .. code-block:: python
+
+            from langchain_community.embeddings import ClovaEmbeddings
+            embeddings = ClovaEmbeddings(
+                clova_emb_api_key='your_clova_emb_api_key',
+                clova_emb_apigw_api_key='your_clova_emb_apigw_api_key',
+                app_id='your_app_id'
+            )
+
+            query_text = "This is a test query."
+            query_result = embeddings.embed_query(query_text)
+
+            document_text = "This is a test document."
+            document_result = embeddings.embed_documents([document_text])
+
+    """
+
+    endpoint_url: str = (
+        "https://clovastudio.apigw.ntruss.com/testapp/v1/api-tools/embedding"
+    )
+    """Endpoint URL to use."""
+    model: str = "clir-emb-dolphin"
+    """Embedding model name to use."""
+    clova_emb_api_key: Optional[SecretStr] = None
+    clova_emb_apigw_api_key: Optional[SecretStr] = None
```

Comment:
could we add docstrings for these

Reply:
I've added docstrings as requested. The changes have been pushed to the branch.
Thank you for your feedback, @baskaryan